lib/kms/file/backend.js

```diff
@@ -0,0 +1,80 @@
+import common from '../common';
+
+export const backend = {
+    /*
+     * Target implementation will be async in some aspects. let mimic it
+     */
+
+    /*
+     * cb(err, masterKeyId: string)
+     */
+    createBucketKey: function createBucketKeyMem(bucketName, log, cb) {
+        process.nextTick(() => {
+            const newKey = common.createDataKey().toString('hex');
+            cb(null, newKey);
+        });
+    },
+
+    /*
+     * cb(err, cipheredDataKey: Buffer)
+     */
+    cipherDataKey: function cipherDataKeyMem(cryptoScheme,
+                                             masterKeyId,
+                                             plainTextDataKey,
+                                             log,
+                                             cb) {
+        process.nextTick(() => {
+            const masterKey = new Buffer(masterKeyId, 'hex');
+            common.createCipher(
+                cryptoScheme, masterKey, 0, log,
+                (err, cipher) => {
+                    if (err) {
+                        cb(err);
+                        return;
+                    }
+                    let cipheredDataKey =
+                            cipher.update(plainTextDataKey);
+                    const final = cipher.final();
```

Comment:
Here to ensure that any bytes remaining in the output of the stream are captured